ingress-inquisition/main.go

```diff
@@ -4,29 +4,83 @@ import (
 	"context"
 	"fmt"
 	"log"
+	"os"
+	"text/tabwriter"
 
 	"github.com/aws/aws-sdk-go-v2/service/ec2"
 	"github.com/aws/aws-sdk-go-v2/service/securityhub"
-	awsauth "github.com/guardian/fsbp-tools/aws-common"
+	"github.com/guardian/fsbp-tools/common"
 	"github.com/guardian/fsbp-tools/ingress-inquisition/utils"
 )
 
+type SecurityGroupAndVpc struct {
+	SecurityGroup string
+	VpcName       string
+	VpcId         string
+}
+
+func getVpcDetails(ctx context.Context, ec2Client *ec2.Client, groupIds []string) []SecurityGroupAndVpc {
+	groupDescriptions, err := ec2Client.DescribeSecurityGroups(ctx, &ec2.DescribeSecurityGroupsInput{
+		GroupIds: groupIds,
+	})
+	if err != nil {
+		log.Fatalf("Error describing security groups: %v", err)
+	}
+
+	res := []SecurityGroupAndVpc{}
+	for _, group := range groupDescriptions.SecurityGroups {
+		vpcs, err := ec2Client.DescribeVpcs(ctx, &ec2.DescribeVpcsInput{
+			VpcIds: []string{*group.VpcId},
+		})
+		if err != nil {
+			log.Fatalf("Error describing VPC: %v", err)
+		}
+		for _, vpc := range vpcs.Vpcs {
+			name := utils.FindTag(vpc.Tags, "Name", "unknown")
+			res = append(res, SecurityGroupAndVpc{
+				SecurityGroup: *group.GroupId,
+				VpcName:       name,
+				VpcId:         *group.VpcId,
+			})
+		}
+	}
+	return res
+}
+
+func findUnusedSecurityGroups(ctx context.Context, ec2Client *ec2.Client, sgIds []string) ([]string, error) {
```

Comment:
This is the only really important bit of code. Everything else is a move, or a change to the way we are logging.